src/fides/api/util/consent_util.py

```diff
@@ -228,3 +247,281 @@ def get_or_create_fides_user_device_id_provided_identity(
+        if privacy_notice.id not in ids:
+            ids.add(privacy_notice.id)
+        else:
+            raise HTTPException(
+                HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=f"More than one provided PrivacyNotice with ID {privacy_notice.id}.",
+            )
+
+
+def create_privacy_notices_util(
+    db: Session,
+    privacy_notice_schemas: List[PrivacyNoticeCreation],
+    should_escape: bool = True,
+) -> Tuple[List[PrivacyNotice], Set[PrivacyNoticeRegion]]:
+    """Performs validation before creating Privacy Notices and Privacy Notice History records
+    and then ensures that Privacy Experiences exist for all Privacy Notices.
+    """
+    validate_notice_data_uses(privacy_notice_schemas, db)  # type: ignore[arg-type]
+
+    existing_notices = PrivacyNotice.query(db).filter(PrivacyNotice.disabled.is_(False)).all()  # type: ignore[attr-defined]
+
+    new_notices = [
+        PrivacyNotice(**privacy_notice.dict(exclude_unset=True))
+        for privacy_notice in privacy_notice_schemas
+    ]
+    check_conflicting_data_uses(new_notices, existing_notices)
+
+    created_privacy_notices: List[PrivacyNotice] = []
+    affected_regions: Set = set()
+
+    for privacy_notice in privacy_notice_schemas:
+        if should_escape:
+            # should_escape flag is for when we're creating a notice
+            # from a template. The content was already escaped in the
+            # template - we don't want to escape twice.
+            privacy_notice = transform_fields(  # type: ignore
+                transformation=escape,
+                model=privacy_notice,
+                fields=PRIVACY_NOTICE_ESCAPE_FIELDS,
+            )
+        created_privacy_notice = PrivacyNotice.create(
+            db=db,
+            data=privacy_notice.dict(exclude_unset=True),
+            check_name=False,
+        )
+        created_privacy_notices.append(created_privacy_notice)
+        affected_regions.update(created_privacy_notice.regions)
+
+    # After creating any new notices, make sure experiences exist to back all notices.
+    upsert_privacy_experiences_after_notice_update(
+        db, affected_regions=list(affected_regions)
+    )
+    return created_privacy_notices, affected_regions
+
+
+def validate_privacy_notice_dry_update(dry_update: PrivacyNotice) -> None:
+    """
+    Verify that a dry update of a PrivacyNotice satisfies the constraints
+    for creating a privacy notice.
+
+    This is done here instead of upfront because we need access to the current values
+    of the privacy notice in the database combined with the patch updates from the request
+    """
+    try:
+        PrivacyNoticeCreation.from_orm(dry_update)
+    except ValueError as exc:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            # pylint: disable=no-member
+            detail=exc.errors(),  # type: ignore
+        )
+
+
+def prepare_privacy_notice_patches(
```

Comment:
This method is copied over from src/fides/api/api/v1/endpoints/privacy_notice_endpoints.py but has the most modifications so it can be shared between updating a privacy notice via the API or _upserting_ a privacy notice _template_ on startup.

It's a little extra logic that can make this confusing to follow but I do like being able to share this code between both usages. There are so many things we want to double check and I want to make sure this happens for both resources.